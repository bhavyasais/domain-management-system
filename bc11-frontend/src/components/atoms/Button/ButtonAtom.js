import React from "react";
import Button from "@material-ui/core/Button";
import { makeStyles, Typography } from "@material-ui/core";
import PropTypes from "prop-types";
import clsx from "clsx";
const useStyles = makeStyles({
  buttonLabel: {
    overflow: "hidden",
    textOverflow: "ellipsis",
  },
});

export default function ButtonAtom({
  value,
  variant,
  disabled,
  color,
  onClick,
  textColor,
  rootStyle,
  labelStyle,
  textVariant,
  maxWidth,
  ...restProps
}) {
  const classes = useStyles();
  return (
    <div>
      <Button
        variant={variant}
        disabled={disabled}
        style={{ backgroundColor: color, maxWidth: maxWidth }}
        onClick={onClick}
        classes={{
          root: rootStyle,
          label: clsx(labelStyle, classes.buttonLabel),
        }}
        {...restProps}
      >
        <Typography variant={textVariant} className={classes.buttonLabel}>
          {value}
        </Typography>
      </Button>
    </div>
  );
}
Button.propTypes = {
  value: PropTypes.string,
  color: PropTypes.string,
  variant: PropTypes.string,
};
Button.defaultProps = {
  variant: "contained",
};
