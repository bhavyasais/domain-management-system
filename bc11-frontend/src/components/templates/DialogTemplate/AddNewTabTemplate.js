import React, { useState } from 'react';
import TextField from '@material-ui/core/TextField';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import ButtonAtom from '../../atoms/Button/ButtonAtom';
import { Typography } from '@material-ui/core';

export default function AddNewTabTemplate(props) {
  const handleAddClick = () => {
    props.onAddClick(tabName);
    props.handleClose();
  };

  const [tabName, setTabName] = useState('');
  const handleTabChange = (e) => {
    setTabName(e.target.value);
  };

  return (
    <div>
      <Dialog open={props.open} onClose={props.handleClose} aria-labelledby="form-dialog-title">
        <Typography variant="h6">Add New Tab</Typography>
        <DialogContent>
          <form>
            <div>
              <TextField
                id="outlined-basic"
                label="Tab Name"
                variant="outlined"
                onChange={handleTabChange}
              />
            </div>
          </form>
        </DialogContent>
        <DialogActions>
          <ButtonAtom
            variant="outlined"
            value="Close"
            textVariant="caption"
            onClick={props.handleClose}
          />
          <ButtonAtom
            variant="contained"
            value="Add"
            textVariant="caption"
            onClick={handleAddClick}
          />
        </DialogActions>
      </Dialog>
    </div>
  );
}
