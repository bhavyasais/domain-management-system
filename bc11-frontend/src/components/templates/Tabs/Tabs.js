import React, { useState } from 'react';
import { Grid, Typography, withStyles } from '@material-ui/core';
import AddIcon from '@material-ui/icons/Add';
import PropTypes from 'prop-types';
import clsx from 'clsx';
import Button from '../../atoms/Button/ButtonAtom';
import AddNewTabTemplate from '../DialogTemplate/AddNewTabTemplate';
import Department from '../../molecules/Department/DepartmentsAdd';

const styles = () => ({
  tabs: {
    backgroundColor: '#eeeeee',
    padding: '10px 32px 10px 32px',
  },
  currentTab: {
    backgroundColor: 'white !important',
    border: '0px !important',
    zIndex: 9999,
    color: '#1c85d8 !important',
    '&:after': {
      content: 'close-quote',
      backgroundColor: 'white !important',
      display: 'inline-block',
      position: 'absolute',
      top: '0px !important',
      border: '0 !important',
    },
    '&:before': {
      content: 'close-quote',
      backgroundColor: 'white !important',
      display: 'inline-block',
      position: 'absolute',
      bottom: '0px !important',
      border: '0 !important',
    },
  },
  tabSpan: {
    display: 'inline-block',
    backgroundColor: '#eaefff',
    padding: '8px 10px 4px 20px',
    borderRadius: '5px 0px 0 0',
    color: '#333',
    border: '1px solid #cccccc',
    cursor: 'pointer',
    textAlign: 'center',
    float: 'left',
    height: '36px',
    position: 'relative',
    boxSizing: 'border-box',
    fontFamily: 'Roboto',
    fontSize: '14px',
    '&:after': {
      width: '16px',
      height: '36px',
      content: 'close-quote',
      backgroundColor: '#eaefff',
      display: 'inline-block',
      position: 'absolute',
      top: '-1px',
      left: '100%',
      border: '1px solid #cccccc',
      borderLeft: 0,
      borderRadius: '0 2px 0 0',
      transform: 'skewX(30deg)',
      boxSizing: 'border-box',
      zIndex: 999,
      borderTopRightRadius: '5px',
    },
    '&:before': {
      width: '10px',
      height: '20px',
      content: 'close-quote',
      backgroundColor: '#eaefff',
      display: 'inline-block',
      position: 'absolute',
      bottom: '-1px',
      left: '100%',
      borderRadius: ' 0 2px 0 0',
      borderBottom: '1px solid #cccccc',
      boxSizing: 'border-box',
      zIndex: 999,
    },
  },
  notFirstChild: {
    marginLeft: '18px',
  },
  button: {
    marginLeft: '24px',
    color: '#1d4cd7',
    '&:hover': {
      backgroundColor: 'transparent',
    },
  },
  contentContainer: {
    width: '100%',
    minHeight: '600px',
    backgroundColor: '#fafafa',
    boxShadow: '0 4px 10px 0 rgba(25, 1, 52, 0.16)',
    borderRadius: '4px',
  },
  labelStyle: {
    color: '#1d4cd7',
    textTransform: 'none',
  },
  tabHeaderContainer: {
    height: '36px',
  },
  footer: {
    width: '100%',
    height: '30px',
    backgroundColor: '#ffffff',
    bottom: 0,
    marginTop: '100px',
    color: '#666666',
    paddingLeft: '12px',
    fontFamily: 'Roboto',
  },
});

const Tabs = ({
  onChange,
  classes,
  tabLabels,
  children: contentArray,
  dept,
  addDepartment,
  addTrustGroup,
  currentDept,
  setCurrentDepartment,
}) => {
  // It's better to handle value hook from the page in which tabs would be used
  const [open, setOpen] = React.useState(false);
  const [selectedRows, setSelectedRows] = useState([]);
  const [totalRows, setTotalRows] = useState([]);
  const [value, setValue] = useState(0);

  const handleClickOpen = () => {
    setOpen(true);
  };

  const handleClose = () => {
    setOpen(false);
  };

  const handleChange = (index) => {
    setValue(index);
    onChange();
  };

  const addColumn = (newTab) => {
    addTrustGroup(newTab);
  };
  return (
    <Grid className={classes.tabs} container direction="column">
      <Grid item className={classes.tabHeaderContainer}>
        {tabLabels &&
          tabLabels.map((label, idx) => (
            <span
              key={idx}
              onClick={() => handleChange(idx)}
              className={clsx(
                classes.tabSpan,
                value === idx && classes.currentTab,
                idx !== 0 && classes.notFirstChild,
              )}
            >
              {label}
            </span>
          ))}
        <Button
          variant="text"
          className={classes.button}
          startIcon={<AddIcon />}
          disableRipple
          value="New tab"
          labelStyle={classes.labelStyle}
          onClick={handleClickOpen}
          textVariant="inherit"
        />
        <AddNewTabTemplate open={open} handleClose={handleClose} onAddClick={addColumn} />
      </Grid>
      <Grid className={classes.contentContainer} container direction="column">
        <Grid item>
          <Department
            dept={dept}
            addDepartment={addDepartment}
            currentDept={currentDept}
            setCurrentDepartment={setCurrentDepartment}
            deleteSelectedRow={selectedRows[value] ?? 0}
          />
        </Grid>
        {contentArray &&
          contentArray.map((content, idx) => (
            <Grid hidden={value !== idx} key={idx} item style={{ flex: 1 }}>
              {React.cloneElement(content, {
                selectedRows,
                setSelectedRows,
                totalRows,
                setTotalRows,
              })}
            </Grid>
          ))}
        <Grid item className={classes.footer} container alignContent="center">
          <Typography variant="body2">
            {selectedRows[value]
              ? `${selectedRows?.[value]} record selected`
              : `${totalRows?.[value]} records`}
          </Typography>
        </Grid>
      </Grid>
    </Grid>
  );
};

Tabs.propTypes = {
  onChange: PropTypes.func,
  tabLabels: PropTypes.arrayOf(PropTypes.string),
  contentArray: PropTypes.arrayOf(PropTypes.element),
  trustGroups: PropTypes.arrayOf(PropTypes.string),
};
Tabs.defaultProps = {
  trustGroups: null,
  onChange: () => {},
  tabLabels: null,
  contentArray: [
    <span>Top 100 content</span>,
    <span>Top 1000 content</span>,
    <span>External partners content</span>,
    <span>Box communication base content</span>,
  ],
};

export default withStyles(styles)(Tabs);
