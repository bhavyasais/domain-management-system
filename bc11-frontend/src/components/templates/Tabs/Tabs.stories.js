import React from "react";
import { withKnobs } from "@storybook/addon-knobs";
import Tabs from "./Tabs";
import DomainGrid from "../../organisms/DomainGrid/DomainGrid";
import {
  DomainrowData,
  trustGroupcolumnDefs,
  trustGrouprowData,
} from "../../../utils/commandCentermockData";
import DataGrid from "../../molecules/DataGrid/DataGrid";
const trustGroups = ["Top 100", "Top 1000", "Top 3000", "Do not trust"];

export default {
  component: Tabs,
  title: "Templates/Tabs",
  decorators: [withKnobs],
};

export const defaulTabs = () => (
  <Tabs
    tabLabels={[
      "Top 100",
      "Top 1000",
      "External partners",
      "Box communication base",
    ]}
    // contentArray={[domainGroup(), domainGroup(), domainGroup(), trustGroup()]}

    contentArray={[
      <DomainGrid rowData={DomainrowData} trustGroups={trustGroups} />,
      <DomainGrid rowData={DomainrowData} trustGroups={trustGroups} />,
      <DomainGrid rowData={DomainrowData} trustGroups={trustGroups} />,
      <DataGrid
        columnDefs={trustGroupcolumnDefs}
        rowData={trustGrouprowData}
      />,
    ]}
  />
);