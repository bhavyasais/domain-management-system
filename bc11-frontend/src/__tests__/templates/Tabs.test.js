import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import { act } from "react-dom/test-utils";
import Tabs from "../../components/templates/Tabs/Tabs";
import AddNewTabTemplate from "../../components/templates/DialogTemplate/AddNewTabTemplate";
import ButtonAtom from "../../components/atoms/Button/ButtonAtom";

describe("<Tabs/>", () => {
  let mounted;
  const onChange = jest.fn();
  beforeEach(() => {
    act(() => {
      mounted = mount(<Tabs onChange={onChange} tabLabels={["tab 1"]} addTrustGroup={()=>{}} handleClose={()=>{}} handleClickOpen={()=>{}}/>);
    });
  });

  it("Renders correctly", () => {
    let wrapper;
    wrapper = renderer.create(<Tabs />);
    expect(wrapper).toMatchSnapshot();
  });
  it("Tab onChange works", () => {
    mounted.find("span").at(0).props().onClick();
    expect(onChange).toHaveBeenCalled();
  });
  it('new tab add', () => {
    mounted.find(AddNewTabTemplate).at(0).props().onAddClick("Top 100");
    mounted.find(AddNewTabTemplate).at(0).props().handleClose();
  });
  it('open a tab', () => {
    mounted.find(ButtonAtom).at(0).props().onClick();
  });
});
