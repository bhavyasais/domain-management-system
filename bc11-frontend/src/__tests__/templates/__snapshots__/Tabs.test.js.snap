// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tabs/> Renders correctly 1`] = `
<div
  className="MuiGrid-root Tabs-tabs-1 MuiGrid-container MuiGrid-direction-xs-column"
>
  <div
    className="MuiGrid-root Tabs-tabHeaderContainer-8 MuiGrid-item"
  >
    <div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text Tabs-button-5"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "backgroundColor": undefined,
            "maxWidth": undefined,
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label Tabs-labelStyle-7 makeStyles-buttonLabel-10"
        >
          <span
            className="MuiButton-startIcon MuiButton-iconSizeMedium"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
              />
            </svg>
          </span>
          <span
            className="MuiTypography-root makeStyles-buttonLabel-10"
          >
            New tab
          </span>
        </span>
      </button>
    </div>
    <div />
  </div>
  <div
    className="MuiGrid-root Tabs-contentContainer-6 MuiGrid-container MuiGrid-direction-xs-column"
  >
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <div
        className="makeStyles-main-13"
      >
        <div
          className="makeStyles-dropDownContainer-11"
          onClick={[Function]}
        >
          <div />
          <div />
        </div>
        <div
          className="makeStyles-items-23"
        >
          <div
            className="makeStyles-iconFilter-21"
          >
            <img
              alt="icon"
              src="filter.png"
            />
            <p
              className="MuiTypography-root makeStyles-textFilter-24 MuiTypography-body1"
            >
              Filter
            </p>
          </div>
          <div
            className="makeStyles-iconFilter-21"
          >
            <img
              alt="icon"
              src="list.png"
            />
            <p
              className="MuiTypography-root makeStyles-textFilter-24 MuiTypography-body1"
            >
              List
            </p>
          </div>
          <div
            className="makeStyles-iconFilter-21"
          >
            <img
              alt="icon"
              src="sort.png"
            />
            <p
              className="MuiTypography-root makeStyles-textFilter-24 MuiTypography-body1"
            >
              Sort
            </p>
          </div>
        </div>
        <div
          className="makeStyles-searchIcon-26"
        >
          <img
            alt="search"
            src="search.png"
          />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root Tabs-footer-9 MuiGrid-container MuiGrid-item MuiGrid-align-content-xs-center"
    >
      <p
        className="MuiTypography-root MuiTypography-body2"
      >
        undefined records
      </p>
    </div>
  </div>
</div>
`;
